core/src/executor/join.rs

```diff
@@ -245,17 +255,42 @@ impl<'a> JoinExecutor<'a> {
             } => (key_expr, value_expr, where_clause),
         };
 
-        let rows_map = storage
-            .scan_data(table_name)
-            .await
-            .map(stream::iter)?
-            .try_filter_map(|(_, row)| {
+        #[derive(futures_enum::Stream)]
+        enum Rows<I1, I2> {
+            Derived(I1),
+            Table(I2),
+        }
+
+        // let blabla: Box<dyn TryStream<Ok = Row, Error = Error>> = match relation {
+        let blabla = match relation {
+            TableFactor::Derived { subquery, alias } => {
+                let filter_context = filter_context.as_ref().map(Rc::clone);
+
+                Rows::Derived(select(storage, subquery, filter_context).await?)
+                // Box::new(select(storage, subquery, filter_context).await?)
+            }
+            TableFactor::Table { .. } => {
+                let rows = storage
+                    .scan_data(relation.get_name()?)
+                    .await?
+                    .map(|item| item.map(|(_, row)| row));
+
+                // Box::new(stream::iter(rows))
+                Rows::Table(stream::iter(rows))
+                /*
+                .map_ok(|(key, row)| row)
+                .map(stream::iter) // .map(|r| r.unwrap().1),
+                */
+            }
```

Comment:
This error was because `select` return result type does not have an associated type of `Item`.
![image](https://user-images.githubusercontent.com/2025065/172060840-54af2e09-0114-45ee-acf3-e16999ae1a64.png)

First, specifying `Item` to `select` and `select_with_labels` are required.
e.g.
```rust
pub async fn select<'a>(                                                       
    storage: &'a dyn GStore,                                                   
    query: &'a Query,                                                          
    filter_context: Option<Rc<FilterContext<'a>>>,                             
) -> Result<impl TryStream<Ok = Row, Error = Error, Item = Result<Row>> + 'a> {
```

Then,
```rust
pub async fn test<'a>(                                                         
    storage: &'a dyn GStore,                                                   
    table_factor: &'a TableFactor,                                             
    filter_context: Option<Rc<FilterContext<'a>>>,                             
) -> Result<impl TryStream<Ok = Row, Error = Error, Item = Result<Row>> + 'a> {
    #[derive(futures_enum::Stream)]                                            
    enum Rows<I1, I2> {                                                        
        Derived(I1),                                                           
        Table(I2),                                                             
    }                                                                          
                                                                               
    match table_factor {                                                       
        TableFactor::Derived { subquery, .. } => {                             
            let filter_context = filter_context.as_ref().map(Rc::clone);       
            let rows = select(storage, subquery, filter_context).await?;       
                                                                               
            Ok(Rows::Derived(rows))                                            
        }                                                                      
        TableFactor::Table { name, .. } => {                                   
            let rows = storage                                                 
                .scan_data(crate::data::get_name(name)?)                       
                .await?                                                        
                .map(|item| item.map(|(_, row)| row));                         
            let rows = stream::iter(rows);                                     
                                                                               
            Ok(Rows::Table(rows))                                              
        }                                                                      
    }                                                                          
}                                                                              
```
These kind of above approach will work.